### PR TITLE
Table perf improvements

### PR DESCRIPTION
**Pull request by larkin**
This includes potential improvements to table performance with large datasets. (~161k rows, many duplicate values in the sort column).

There are three different changes in this PR:

A. defer dupe check
B. shortcut insertion check
C. remove format/parse cache

test case | processData elapsed
 --- | --- 
baseline | 87s 
A | 26s 
A+B | 16s 
A+B+C | 13s 

I'd consider the 3 seconds to maintain the format/parse cache a non-negligible hit; let this be a lesson to myself as to why premature optimization is bad.  The bulk of the remaining performance hit is moment date parsing.

Note that there's also 45-60 seconds of network latency loading this table.

@akhenry would love your insights on these changes before I update tests.

### CODE REVIEW

File: platform/features/table/src/controllers/TelemetryTableController.js [hunk 1]
```diff
@@ -291,6 +291,7 @@ define(
                  */
                 function finishProcessing() {
                     telemetryCollection.add(rowData);
+                    telemetryCollection.dupeCheck = true;
```
Comment (akhenry) [inline platform/features/table/src/controllers/TelemetryTableController.js]: Let's move this into `TelemetryCollection.add`

**Comment by larkin** [comment]
Failing test was due to a bug in my code, resolved that.  And moved code to TelemetryCollection.add.

**Comment by larkin** [comment]
# Author Checklist

1. Changes address original issue? Y (as included here)
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

**Comment by akhenry** [comment]
@larkin what's the status of this? Can we merge it?

**Comment by larkin** [comment]
@akhenry this just requires a review and it's good to merge.

**Comment by VWoeltjen** [comment]
### Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? N (performance improvement covered by existing tests)
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y

**Pull request merged**